pkg/overlay/config.go

```diff
@@ -72,11 +74,19 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (
 		return err
 	}
 
+	ticker := time.NewTicker(time.Duration(c.RefreshInterval))
+	refreshErrors := make(chan error)
+
 	go func() {
-		// TODO(jt): should there be a for loop here?
-		err := cache.Refresh(ctx)
-		if err != nil {
-			zap.S().Fatal("cache refreshes stopped", zap.Error(err))
+		for {
+			select {
+			case <-ticker.C:
+				err := cache.Refresh(ctx)
+				if err != nil {
+					zap.S().Error("Error with cache refresh")
+					refreshErrors <- err
```

Comment:
refreshErrors is an unbuffered go channel, which means that because no one is reading off of it, this call will block forever. i'd prefer just forgetting about the channel instead of trying to fix this problem

Reply:
Okay, dropped the refresh channel, I have it so it is just logging errors now, which will happen on every Tick.